src/test/java/io/confluent/examples/streams/GenericAvroIntegrationTest.java

```diff
@@ -95,12 +97,12 @@ public void shouldRoundTripGenericAvroDataThroughKafka() throws Exception {
     // However, in the code below we intentionally override the default serdes in `to()` to
     // demonstrate how you can construct and configure a generic Avro serde manually.
     final Serde<String> stringSerde = Serdes.String();
-    final Serde<GenericRecord> genericAvroSerde = new MockGenericAvroSerde(mockSchemaRegistryClient);
-    // Note how we must manually call `configure()` on this serde to configure the schema registry
-    // url.  This is different from the case of setting default serdes (see `streamsConfiguration`
+    final Serde<GenericRecord> genericAvroSerde = new GenericAvroSerde();
+    // Note how we must manually call `configure()` on this serde to configure the schema registry.
+    // This is different from the case of setting default serdes (see `streamsConfiguration`
     // above), which will be auto-configured based on the `StreamsConfiguration` instance.
     genericAvroSerde.configure(
-        Collections.singletonMap(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, "dummy config"),
+        Collections.singletonMap(AbstractKafkaAvroSerDeConfig.MOCK_SCHEMA_REGISTRY_CONFIG, SCHEMA_REGISTRY_SCOPE),
```

Comment:
Now, instead of passing in your own "mock registry" subclass, you just pass in your desired scope name. This is actually a very similar concept to the schema registry url parameter, since the scope name uniquely references a single (mocked) Schema Registry instance.